test/test_segmentation.cpp

```diff
@@ -282,4 +291,42 @@ TEST_F(SegmentationTest, InputForwardingUntilOutput) {
       fec.fusion(), out_tensors, {in_tensor, in_tensor}, __LINE__, __FILE__);
 }
 
+TEST_F(SegmentationTest, ForwardedExprsAreNotMergeable) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  auto tv0 = makeSymbolicTensor(1);
```

Comment:
Sorry, I missed this on first review. This does not repro #1541 exactly since that issue used half precision inputs, which will behave differently in segmentation due to `castInputOutputToLowerPrecision()`.